wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorServiceImpl.java

```diff
@@ -89,7 +94,12 @@ public MonitorConnectionContext startMonitoring(
           () -> Messages.get(
               "MonitorServiceImpl.emptyAliasSet",
               new Object[] {hostSpec}));
-      hostSpec.addAlias(hostSpec.asAlias());
+      try {
+        this.pluginService.fillAliases(connectionToAbort, hostSpec);
```

Comment:
That's a tricky part. This method has nodeKeys parameter that should be used to identify a proper Monitor. At the same time we have hostSpec that we fill with aliases. We need to decide which way we want to identify a proper Monitor.